src/common_utils.c

```diff
@@ -201,35 +201,43 @@ bool amountToString(const uint8_t *amount,
                     const char *ticker,
                     char *out_buffer,
                     size_t out_buffer_size) {
-    char tmp_buffer[100] = {0};
+    uint8_t str_len = 0;
+    uint8_t amount_len = 0;
+    uint8_t ticker_len = 0;
+    char tmp_buffer[50 + MAX_TICKER_LEN] = {0};  // 50 is sufficient for the amount itself
 
+    memset(out_buffer, 0, out_buffer_size);
+
+    // Convert the Amount to decimal string first
     if (uint256_to_decimal(amount,
                            amount_size,
                            tmp_buffer,
                            sizeof(tmp_buffer)) == false) {
+        PRINTF("uint256_to_decimal failed\n");
         return false;
     }
-
-    uint8_t amount_len = strnlen(tmp_buffer, sizeof(tmp_buffer));
-    uint8_t ticker_len = strnlen(ticker, MAX_TICKER_LEN);
-
-    if (ticker_len > 0) {
-        if (out_buffer_size <= ticker_len + 1) {
-            return false;
-        }
-        memcpy(out_buffer, ticker, ticker_len);
-        out_buffer[ticker_len++] = ' ';
-    }
-
+    // Adjust the decimal position, store the result in out_buffer
+    amount_len = strnlen(tmp_buffer, sizeof(tmp_buffer));
+    ticker_len = strnlen(ticker, MAX_TICKER_LEN) + 1;  // + 1 for \0
+    str_len = out_buffer_size - ticker_len - 1;  // - 1 for space
```

Comment:
All this length and size are getting complicated to follow
If we remove MAX_TICKER_LEN from tmp_buffer we can forward the sizeof(tmp_buffer) here